### PR TITLE
RA-1517 Integrating MPI Functionality

### PR DESCRIPTION
**Pull request by shaoyuancc**
Hi All,

This merge is meant to bring in MPI functionality primarily developed by SolDevlo for iSantePlus back into the master branch. 

Much of the discussion is recorded here: https://talk.openmrs.org/t/integration-with-openempi/11535/45

Jira Ticket: https://issues.openmrs.org/browse/RA-1517

### CODE REVIEW

**Comment by slubwama** [comment]
@shaoyuancc rename your PR to reflect the issue ID in JIRA

**Comment by shaoyuancc** [comment]
Squashed!

**Comment by slubwama** [comment]
@dkayiwa  Yes. I can not pull a PR in my local PR

**Comment by dkayiwa** [comment]
These two commands are all that you need:

git checkout -b shaoyuancc-master master
git pull https://github.com/shaoyuancc/openmrs-module-coreapps.git master

**Comment by shaoyuancc** [comment]
Hi, I am so sorry for the delayed responses, i've been really busy. It was ready when i first submitted the pull request at the end of July, but now that other changes have been added to coreapps since then i havent tested it with these changes pulled... and i don't think i would have time to do it until the end of december :(

**Comment by dkayiwa** [comment]
@shaoyuancc do you think we should merge this for now and do a follow up pull request if more changes are needed? Or would you rather we leave it?

**Comment by shaoyuancc** [comment]
Hi @dkayiwa yes, i think we should go ahead and merge...

File: omod/src/main/webapp/resources/scripts/patientsearch/patientSearchWidget.js [hunk 1]
```diff
@@ -256,9 +262,12 @@ function PatientSearchWidget(configuration){
                 }
                 var identifier = patient.patientIdentifier != null ? patient.patientIdentifier.identifier : null;
                 if(_.contains(initialPatientUuids, patient.uuid)){
-                    identifier = patient.patientIdentifier.identifier+
```
Comment (dkayiwa) [inline omod/src/main/webapp/resources/scripts/patientsearch/patientSearchWidget.js]: Before i merge, did you intentionally drop this? identifier = patient.patientIdentifier.identifier

**Pull request merged**